### ISSUE TITLE
7GUIs circle drawer example does not work

### ISSUE THREAD

**Issue by GammaSagittarii** [vizia/vizia#471]
Currently dialog widget is not a popup nor a window, but a widget that is fully transparent until shown – which means its size is counted in the layout as if it is always shown. I assume this is just a temporary implementation, and will be changed in the future.
In case I am wrong about it being current expected behavior, you can see what I mean by running 7GUIs circle drawer example. It can be seen in the save dialog example too, it is just not as obvious.

On the left is the current state of  the circle drawer example, on the right is version with fixes that restore behavior from the initial commit.

![sidebyside](https://github.com/vizia/vizia/assets/32195338/db8c6d14-1cd7-4038-9957-3e45598c569f)

Additional issues in circle drawer example:
1. Popup menu is not triggering dialog to show - as the on press event is attached to the popup instead of the button in the popup.
2. If you attach the on press event to the button, the dialog does not show coordinates of the circle it looks like this: 
![dialog](https://github.com/vizia/vizia/assets/32195338/4439fcc9-8cbd-47e9-94c1-638e5735c2d2)

3. The dialog is not closed upon clicking to draw more circles, because  it does not have an on blur event.

I have restored previous behavior on my branch fix_circle_drawer. I can make a pull request if you want, but I suspect you’d like to wait with fixes until dialog widget has been improved.

**Comment by geom3trik** [comment]
You are right, this example was broken by the last major update. I removed the `Dialog` view as I wasn't sure if a dedicated view for it was necessary. I was going to replace it with something but forgot, so the example was broken. I'm happy to take a look at your implementation or hear any ideas you have for how you think dialogs should work.

**Comment by GammaSagittarii** [comment]
Just to clarify, the issue seems to be that setting bottom property does nothing if the top is not set.

**Comment by geom3trik** [comment]
> Just to clarify, the issue seems to be that setting bottom property does nothing if the top is not set.

Oh yeh bottom won't work unless top is set to stretch. Can also work with auto top, which it is by default, but only if the parent sets child-top to stretch.
